### PR TITLE
#1852/Shortening token symbols

### LINKED ISSUE #1852: Enhance Limit order form when token name is long
Reported in https://github.com/cowprotocol/cowswap/pull/1843#issuecomment-1378742950

To reproduce: On limit orders, load token https://etherscan.io/token/0x2E2A42fbE7C7E2FfC031BAF7442dBE1F8957770a
See the form:
image
image

**Expected**: Enhance Limit order form when token name is long

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

## Updated 2023-01-24

I really hope now I got all the places 🤞 
Checked Activity modals, pop-ups, swap and limit orders

Screenshot 2023-01-24 at 15 54 49
Screenshot 2023-01-24 at 15 54 38
Screenshot 2023-01-24 at 15 49 17
Screenshot 2023-01-24 at 15 34 17
Screenshot 2023-01-24 at 15 34 06

## Testing

I've added a temporary code to make every new token added suuuuper long
Will of course revert before merging.

1. Add a token that's not in any list. For example, `0x07865c6e87b9f70255377e024ace6630c1eaa37f` on Goerli
2. Swap, limit, anything you can think of with this
* Name should be trimmed everywhere, except in titles

---

## Original

Closes #1852 

Follow up to https://github.com/cowprotocol/cowswap/pull/1843

Screenshot 2023-01-11 at 15 34 35

# To Test

1. Load token https://etherscan.io/token/0x2E2A42fbE7C7E2FfC031BAF7442dBE1F8957770a
4. Check both swap and limit orders
* The ui should not be broken

Note: further testing required in all part of the app. For that would be nice to find a georli token with a long symbol name.
Not able to test with that one on mainnet as there's no liquidity

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @alfetopito , I will try to find a token with a long name in gorli or GC, but for now there are 2 places where it would be great to adjust the token name: 
limit price
price
Favoutrites list: 
image

Also, I think that it would be great not to crop the token name in title 
image

WDYT about adding a title with the token name here? 
image

**Comment by elena-zh** [comment]
Actually, I can't find a token with a long name in Gorli or GC, so I played with existing data and noticed that there are few places left where the long token name can brake the UI: 
image

**Review by anxolin — APPROVED** [review]

File: src/cow-react/utils/format.ts [hunk 1]
```diff
@@ -191,3 +191,9 @@ export function truncateOnMaxDecimals(value: string, decimals: number): string {
   const regex = new RegExp(`(\\d*\\.\\d{${decimals}})\\d*`)
   return value.replace(regex, '$1')
 }
+
+const MAX_SYMBOL_LENGTH = 8
+
+export function formatSymbol(symbol: string | undefined): string | undefined {
```
Comment (shoom3301) [inline src/cow-react/utils/format.ts:197]: It works great, however, would be cool to create a component for this.
Like:
```
function TokenSymbol({token}){
    return <span title={token.symbol}>{formatSymbol(token.symbol)}</span>
}
```

**Review by shoom3301 — APPROVED** [review]

File: src/cow-react/utils/format.ts (hunk 1 above)
Comment (alfetopito) [reply src/cow-react/utils/format.ts]: Great idea, I've updated it

**Comment by elena-zh** [comment]
Hey @fairlighteth , all looks good besides Receive/From tooltips:
 - on the Swap form: word 'before' is wrapped
 
tooltip form
 - on the order confirm modal:
 
order confirm modal

**Comment by alfetopito** [comment]
> Hey @fairlighteth , all looks good besides Receive/From tooltips:
> 
>     * on the Swap form: word 'before' is wrapped
> 
> 
> tooltip form
> 
>     * on the order confirm modal:
> 
> 
> order confirm modal

@elena-zh should be fixed now (famous last words)

**Comment by elena-zh** [comment]
> > Hey @fairlighteth , all looks good besides Receive/From tooltips:
> > ```
> > * on the Swap form: word 'before' is wrapped
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > tooltip form
> > ```
> > * on the order confirm modal:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > order confirm modal
> 
> @elena-zh should be fixed now (famous last words)

Hey @alfetopito , no, not fixed :(
Swap form: 
before

Order confirm modal
modal

**Comment by elena-zh** [comment]
Hey @alfetopito , great job!!!
Just 1 issue: token name is weird in the Approve token modal: 
approving
 And 1 nitpick: there is a bigger indent between the last word and a token name on the Approve button:
штвуте

**Comment by alfetopito** [comment]
> Hey @alfetopito , great job!!! Just 1 issue: token name is weird in the Approve token modal: 

Thank you Elena, both fixed

**Review by elena-zh — APPROVED** [review]
Yeeea!! Approved now!

**Comment by alfetopito** [comment]
I forgot to revert the temporary change added in this PR for testing before merging it to `develop`.

Pushed a revert on commit 34e005778514929c2f68498855e1f08e8570d052

**Pull request merged**